include/GIL/InstBase.hpp

```diff
@@ -4,18 +4,60 @@
 
 #include <llvm/ADT/ArrayRef.h>
 #include <llvm/ADT/PointerUnion.h>
+#include <llvm/ADT/APInt.h>
+#include <llvm/ADT/APFloat.h>
+#include <llvm/ADT/StringRef.h>
 
 namespace glu::gil {
 
-class GILValue;
-class GILBasicBlock;
+class InstBase;
+class BasicBlock {std::string label;}; // FIXME: Placeholder
 
 enum class InstKind {
     #define GIL_INSTRUCTION(CLS, NAME) CLS ## Kind,
     #include "InstKind.def"
     #undef GIL_INSTRUCTION
 };
 
+/**
+ * @class GILValue
+ * @brief Represents a value reference in the GIL (Glu Intermediate Language).
+ *
+ * This class encapsulates a value that is either defined by an instruction or is an argument of a basic block.
+ * It provides methods to retrieve the defining instruction, the defining block, and the index of the value.
+ * 
+ * Example GIL code:
+ * @code
+ * entry(%0, %1):
+ *  %2 = add %0, %1
+ *  return %2
+ * @endcode
+ * 
+ * In this example, %0 = GILValue(entry, 0), %1 = GILValue(entry, 1), and %2 = GILValue(add, 0). %0 and %1 are
+ * basic block arguments, while %2 is a result of the add instruction. The return instruction has no results.
+ * Values are given indices in the order they are defined, at the time of printing the GIL code. The indices are
+ * not stored anywhere in the GIL.
+ */
+class GILValue {
```

Comment:
Consider a new name. VirtualReg? SSAValue? Value? Register?

Reply:
SSAValue is the most precise one but Value is the easiest to understand.
I think we should name it SSAValue

Reply:
Value suits me (LLVM uses Value I think), but as you wish :)

Reply:
Value it is